packages/datadog-plugin-http/src/client.js

```diff
@@ -145,45 +145,80 @@ function patch (http, methodName, tracer, config) {
     })
   }
 
-  function extractUrl (options) {
-    const uri = options
-    const agent = options.agent || http.globalAgent
-
-    return typeof uri === 'string' ? uri : url.format({
-      protocol: options.protocol || agent.protocol,
-      hostname: options.hostname || options.host || 'localhost',
-      port: options.port,
-      pathname: options.path || options.pathname || '/'
-    })
-  }
-
   function normalizeArgs (inputURL, inputOptions, callback) {
-    let options = normalizeURL(inputURL)
+    inputURL = normalizeOptions(inputURL)
+
+    const [callbackNormalilzed, inputOptionsNormalized] = normalizeCallback(inputOptions, callback, inputURL)
+    const coalesedOptions = coaleseOptions(inputURL, inputOptionsNormalized)
+    const uri = url.format(coalesedOptions)
 
+    return { uri: uri, options: coalesedOptions, callback: callbackNormalilzed }
+  }
+
+  function normalizeCallback (inputOptions, callback, inputURL) {
     if (typeof inputOptions === 'function') {
-      callback = inputOptions
-    } else if (typeof inputOptions === 'object') {
-      options = Object.assign(options, inputOptions)
+      return [inputOptions, inputURL || {}]
+    } else {
+      return [callback, inputOptions]
     }
-    const uri = extractUrl(options)
-    return { uri, options, callback }
   }
 
-  function normalizeURL (inputURL) {
-    let options
+  function coaleseOptions (inputURL, inputOptions) {
+    let coalesedOptions
 
-    if (typeof inputURL === 'string') {
-      options = url.parse(inputURL)
+    if (typeof inputOptions === 'object') {
```

Comment:
Extract the condition to a function to avoid using `let`.

Reply:
updated